component/vdr/indy/vdr.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright Scoir Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package indy implements a VDR interface for Aries (aries-framework-go).
+//
+package indy
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	vdrapi "github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+	"github.com/hyperledger/indy-vdr/wrappers/golang/vdr"
+)
+
+// Client is the API for interacting with the Indy VDR client.
+type Client interface {
+	GetNym(did string) (*vdr.ReadReply, error)
+	GetEndpoint(did string) (*vdr.ReadReply, error)
+	GetPoolStatus() (*vdr.PoolStatus, error)
+	RefreshPool() error
+	Close() error
+}
+
+// VDR represents a Verifiable Data Registry for use with Indy ledgers.
+type VDR struct {
+	MethodName string
+	Refresh    bool
+	Client     Client
+}
+
+// New creates an instance of an Indy VDR.
+func New(methodName string, opts ...Option) (*VDR, error) {
+	vdri := &VDR{MethodName: methodName}
+
+	for _, opt := range opts {
+		opt(vdri)
+	}
+
+	if vdri.Client == nil {
+		return nil, errors.New("an Indy Ledger client must be set with an option to New")
+	}
+
+	if vdri.Refresh {
+		err := vdri.Client.RefreshPool()
+		if err != nil {
+			return nil, fmt.Errorf("refreshing indy pool failed: %w", err)
+		}
+	}
+
+	return vdri, nil
+}
+
+// Store is a no-op for the Indy VDR.
+func (r *VDR) Store(doc *did.Doc, by *[]vdrapi.ModifiedBy) error {
+	return nil
+}
+
+// Accept the did method.
+func (r *VDR) Accept(method string) bool {
+	return method == r.MethodName
+}
+
+// Close the client connection.
+func (r *VDR) Close() error {
+	return r.Client.Close()
+}
+
+// Option configures the Indy vdri.
+type Option func(opts *VDR)
+
+// WithRefresh option configuring an initial refresh of the Indy pool.
+func WithRefresh(refresh bool) Option {
+	return func(opts *VDR) {
+		opts.Refresh = refresh
+	}
+}
+
+// WithIndyClient configures the VDR with an existing Indy Client.
+func WithIndyClient(client Client) Option {
```

Comment:
Resolution seems to depend entirely on this client. Will an implementation be provided soon?

Reply:
Yes, that client is implemented in https://github.com/hyperledger/indy-vdr/tree/master/wrappers/golang